tests/solana_tests/using.rs

```diff
@@ -92,6 +126,67 @@ fn using_for_contracts() {
         .call();
 
     assert_eq!(runtime.logs, "X libX contractx:2");
+
+    let mut runtime = build_solidity(
+        r#"
+        interface I {
+            function f1(int) external;
+            function X(int) external;
+        }
+
+        library L {
+            using {L.X} for I;
+
+            function f1_2(I i) external {
+                i.f1(2);
+            }
+
+            function X(I i) external {
+                print("X lib");
+            }
+
+            function Y(I i) external {
+                i.X();
+            }
```

Comment:
Would you mind not passing the interface as a parameter?
#1517 (awaiting your review) will make this stop working.

Reply:
The function `using_for_contracts` is completely removed in that PR because it is not possible to test libraries for interfaces.

Reply:
That's not good

Reply:
> That's not good

Well, if an interface cannot be a type, then there cannot be a library for it.

Reply:
Let's deal with #1517  before we deal with this